eng/test-rebuild.ps1

```diff
@@ -27,9 +29,21 @@ try {
   . (Join-Path $PSScriptRoot "build-utils.ps1")
   Push-Location $RepoRoot
 
-  Write-Host "Building Roslyn"
-  Exec-Console (Join-Path $PSScriptRoot "build.ps1") "-restore -build -ci:$ci -configuration:$configuration -pack -binaryLog"
-  Exec-Console "artifacts\bin\BuildValidator\$configuration\net472\BuildValidator.exe" "--assembliesPath '$ArtifactsDir/obj/Microsoft.CodeAnalysis'"
+  if (-not $noBuild) {
+    Write-Host "Building Roslyn"
+    Exec-Block { & (Join-Path $PSScriptRoot "build.ps1") -build -bootstrap -ci:$ci -configuration:$configuration -pack -binaryLog }
+  }
+
+  $dotnetInstallDir = (InitializeDotNetCli -install:$true)
+  $rebuildArgs = ("--verbose" +
+  " --assembliesPath `"$ArtifactsDir/obj/Microsoft.CodeAnalysis/$configuration`"" +
+  " --assembliesPath $ArtifactsDir/obj/csc/$configuration/netcoreapp3.1" +
+  " --debugPath `"$ArtifactsDir/BuildValidator`"" +
+  " --sourcePath `"$RepoRoot`"" +
+  " --referencesPath `"$ArtifactsDir/bin`"" +
+  " --referencesPath `"$dotnetInstallDir/packs/Microsoft.AspNetCore.App.Ref`"" +
```

Comment:
Extracting these folders out to command line arguments was necessary to build on Helix agents where dotnet is typically not found under `C:\Program Files\dotnet`.